src/ngmca.js

```diff
@@ -0,0 +1,96 @@
+import { Matrix } from 'ml-matrix';
+
+import {
+  checkMatrixS,
+  initialize,
+  normalize,
+  updateLambda,
+  updateMatrixA,
+  updateMatrixS,
+} from './stages';
+
+/**
+ * Performing non-negative matrix factorization solving argmin_(A >= 0, S >= 0) 1 / 2 * ||Y - AS||_2^2 + lambda * ||S||_1
+ * @param {*} originalMatrix - Matrix to be separated.
```

Comment:
Should not be '*' what it may be ? Matrix||Array[][] ?

Reply:
Matrix||Array<Array> is it ok?